File: backend/src/main/java/com/tribly/dto/publications/response/PublicationListResponse.java
package com.tribly.dto.publications.response;

import com.tribly.dto.validation.ValidateSchema;
import java.util.List;
import org.eclipse.microprofile.openapi.annotations.media.Schema;

@Schema(description = "Paginated publication list response")
@ValidateSchema
public record PublicationListResponse(
    @Schema(description = "List of publications", required = true)
        List<PublicationDto> publications,
    @Schema(description = "Total number of publications", required = true) long total,
    @Schema(description = "Current page number", required = true) int page,
    @Schema(description = "Page size", required = true) int size) {}

File: frontend/src/api/dto/publicationListResponse.ts
import type { PublicationDto } from './publicationDto.ts'

/**
 * Paginated publication list response
 */
export interface PublicationListResponse {
  /** List of publications */
  publications: PublicationDto[]
  /** Total number of publications */
  total: number
  /** Current page number */
  page: number
  /** Page size */
  size: number
}
